### ISSUE TITLE
New recipe Changes back during runtime

### ISSUE THREAD

**Issue by justbtoo** [huika/UMA#54]
I've added 2 separate shared colors color 0 and Color 1.

I've assigned color 0 to one overlay and color 1 to another. When I press play the character generates and the settings change both overlays to color 0. Plus another overlay gets check marked after unchecking it.

I've also used save collection

![screenhunter_1303 oct 30 12 39](https://cloud.githubusercontent.com/assets/12765338/10851971/7fce32c4-7f04-11e5-8fac-2b538e4227dd.jpg)
![screenhunter_1304 oct 30 12 40](https://cloud.githubusercontent.com/assets/12765338/10851974/878caaf4-7f04-11e5-83b4-413d5fe3a443.jpg)
![screenhunter_1305 oct 30 12 40](https://cloud.githubusercontent.com/assets/12765338/10851975/8792b57a-7f04-11e5-9583-70a0223d9c9f.jpg)
![screenhunter_1306 oct 30 12 40](https://cloud.githubusercontent.com/assets/12765338/10851976/8795a712-7f04-11e5-80af-09fdb52a3ce6.jpg)
![screenhunter_1307 oct 30 12 40](https://cloud.githubusercontent.com/assets/12765338/10851977/879811f0-7f04-11e5-87eb-7de61fa87bc0.jpg)
![screenhunter_1308 oct 30 12 40](https://cloud.githubusercontent.com/assets/12765338/10851978/879b20b6-7f04-11e5-8453-d4b329d8fdde.jpg)

**Comment by Jaimi** [comment]
Probably a remnant of the code left that tried to do it all automatically
for you.. I thought I got all those. Ill go back and verify. Are you using
the version from the repository?

**Comment by justbtoo** [comment]
Thanks Jaimi.

I'm using the one from the store. 2.0.5

**Comment by Jaimi** [comment]
Ok - I think I found the line of code that was resetting these. I'll check
in the code to fix it here in a few minutes, but if you want to fix your
version, the change is minor -

On or about line 648 in CharacterBaseEditor.cs, change this section of code:

         if (_recipe.sharedColors != null)
         {
            _sharedColors = ArrayUtility.Contains<OverlayColorData>(_recipe.
sharedColors, _overlayData.colorData);
         }

To this:

         if (_recipe.sharedColors != null)
         {
            foreach(OverlayColorData ocd in _recipe.sharedColors)
            {
               if (ocd.GetHashCode() == _overlayData.GetHashCode()
_sharedColors = true;
            }
         }


Basically what was happening is that ArrayUtility.Contains was comparing
the colors based on the values of the colors, and this was erroneously
returning that the shared color was found (when it didn't exist).

**Comment by Jaimi** [comment]
Fixed in 0093963
